tests/end_to_end/test_create_new_template.py

```diff
@@ -0,0 +1,188 @@
+import datetime
+import os
+import re
+import uuid
+
+from playwright.sync_api import expect
+
+E2E_TEST_URI = os.getenv("NOTIFY_E2E_TEST_URI")
+
+
+def _setup(page, end_to_end_context):
+    # Prepare for adding a new service later in the test.
+    current_date_time = datetime.datetime.now()
+    new_service_name = "E2E Federal Test Service {now} - {browser_type}".format(
+        now=current_date_time.strftime("%m/%d/%Y %H:%M:%S"),
+        browser_type=end_to_end_context.browser.browser_type.name,
+    )
+
+    page.goto(f"{E2E_TEST_URI}/accounts")
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    # Check to make sure that we've arrived at the next page.
+    # Check the page title exists and matches what we expect.
+    expect(page).to_have_title(re.compile("Choose service"))
+
+    # Check for the sign in heading.
+    sign_in_heading = page.get_by_role("heading", name="Choose service")
+    expect(sign_in_heading).to_be_visible()
+
+    # Retrieve some prominent elements on the page for testing.
+    add_service_button = page.get_by_role(
+        "button", name=re.compile("Add a new service")
+    )
+
+    expect(add_service_button).to_be_visible()
+
+    existing_service_link = page.get_by_role("link", name=new_service_name)
+
+    # Check to see if the service was already created - if so, we should fail.
+    # TODO:  Figure out how to make this truly isolated, and/or work in a
+    #        delete service workflow.
+    expect(existing_service_link).to_have_count(0)
+
+    # Click on add a new service.
+    add_service_button.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    # Check for the sign in heading.
+    about_heading = page.get_by_role("heading", name="About your service")
+    expect(about_heading).to_be_visible()
+
+    # Retrieve some prominent elements on the page for testing.
+    service_name_input = page.locator('xpath=//input[@name="name"]')
+    add_service_button = page.get_by_role("button", name=re.compile("Add service"))
+
+    expect(service_name_input).to_be_visible()
+    expect(add_service_button).to_be_visible()
+
+    # Fill in the form.
+    service_name_input.fill(new_service_name)
+
+    # Click on add service.
+    add_service_button.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    # Check for the service name title and heading.
+    service_heading = page.get_by_text(new_service_name, exact=True)
+
+    expect(service_heading).to_be_visible()
+    expect(page).to_have_title(re.compile(new_service_name))
+
+    return new_service_name
```

Comment:
I like what you did here with a `_setup` method (and corresponding `_teardown()` method)!

Seeing this now across a few tests I wonder if we shift this into a more generalized `_setup()` or `_setup_service()` method that lives in `tests/end_to_end/conftest.py` since this will likely be a common step in most tests yet we'll want to keep separate for each one for isolation purposes.

I don't think this is something we need to change in this PR or any of the others currently open, but something to consider going forward!